profiler/models/profiler_profile.py

```diff
@@ -0,0 +1,470 @@
+                            "ZONE 'UTC', 'YYYY-MM-DD HH24:MI:SS')")
+        now = self.env.cr.fetchall()[0][0]
+        # now = fields.Datetime.to_string(
+        #     fields.Datetime.context_timestamp(self, datetime.now()))
+        return now
+
+    @api.multi
+    def enable(self):
+        self.ensure_one()
+        if tools.config.get('workers'):
+            raise exceptions.UserError(
+                _("Start the odoo server using the parameter '--workers=0'"))
+        _logger.info("Enabling profiler")
+        self.write(dict(
+            date_started=self.now_utc(),
+            state='enabled'
+        ))
+        ProfilerProfile.enabled = self.enable_python
+        self._reset_postgresql()
+
+    @api.multi
+    def _reset_postgresql(self):
+        if not self.enable_postgresql:
+            return
+        if ProfilerProfile.pglogs_enabled:
+            _logger.info("Using postgresql.conf or PGOPTIONS predefined.")
+            return
+        os.environ['PGOPTIONS'] = (
+            PGOPTIONS_ENV if self.state == 'enabled' else '')
+        self._reset_connection(self.state == 'enabled')
+
+    def _reset_connection(self, enable):
+        for connection in sql_db._Pool._connections:
+            with connection[0].cursor() as pool_cr:
+                params = (PGOPTIONS if enable
+                          else ProfilerProfile.psql_params_original)
+                for param, value in params.items():
+                    try:
+                        pool_cr.execute('SET %s TO %s' % (param, value))
+                    except (OperationalError, ProgrammingError) as oe:
+                        pool_cr.connection.rollback()
+                        raise exceptions.UserError(
+                            _("It's not possible change parameter.\n%s\n"
+                              "Please, disable postgresql or re-enable it "
+                              "in order to read the instructions") % str(oe))
+            ProfilerProfile.activate_deactivate_pglogs = enable
+
+    def get_stats_string(self, cprofile_path):
+        pstats_stream = StringIO()
+        pstats_obj = pstats.Stats(cprofile_path, stream=pstats_stream)
+        pstats_obj.sort_stats('cumulative')
+        pstats_obj.print_stats()
+        pstats_stream.seek(0)
+        stats_string = pstats_stream.read()
+        pstats_stream = None
+        return stats_string
+
+    @api.multi
+    def dump_postgresql_logs(self, indexed=None):
+        self.ensure_one()
+        self.description = ''
+        pgbadger_cmd = self._get_pgbadger_command()
+        if pgbadger_cmd is None:
+            return
+        pgbadger_cmd_str = subprocess.list2cmdline(pgbadger_cmd)
+        self.description += (
+            '\nRunning the command: %s') % pgbadger_cmd_str
+        result = tools.exec_command_pipe(*pgbadger_cmd)
+        datas = result[1].read()
+        if not datas:
+            self.description += "\nPgbadger output is empty!"
```

Comment:
AFAIK each `self.description +=` will issue a `self.write()` from the ORM. Isn't it better to just use a normal variable and assign only once at the end?